libs/render_delegate/render_delegate.h

```diff
@@ -573,8 +585,12 @@ class HdArnoldRenderDelegate final : public HdRenderDelegate {
     /// Method used to create any node in the context of the render delegate. 
     /// This method should always be called, instead of explicit AiNode() creations
     inline 
-    AtNode * CreateArnoldNode(const AtString &nodeType, const AtString &nodeName) {
-        AtNode *node = AiNode(GetUniverse(), nodeType, nodeName, _procParent);
+    AtNode * CreateArnoldNode(const AtString &nodeType, const AtString &nodeName, bool checkExisting = false) {
```

Comment:
Can we have a function FindOrCreateArnoldNode instead of adding a `checkExisting` argument ?